src/pybind/mgr/ssh/module.py

```diff
@@ -684,3 +683,22 @@ def update_mgrs(self, num, hosts):
             results.append(result)
 
         return SSHWriteCompletion(results)
+
+    def add_mds(self, spec):
+        host = self._get_random_host()
```

Comment:
Let's keep this thing predictable. If you want your tool to automatically decide where to deploy services, please use Kubernetes. Otherwise things like resource usage, resource requirements or taints will creep in eventually.